app/src/main/java/com/github/se/orator/model/chatGPT/ChatViewModel.kt

```diff
@@ -516,7 +517,117 @@ The session is now over. According to the initial instructions, you can now brea
     getNextGPTResponse()
   }
 
-  private fun messagesToTranscript(messages: List<Message>): String {
-    return messages.filter { it.role == "user" }.joinToString("\n") { it.content }
+  /**
+   * Evaluates the battle candidates and provides feedback to the winner and loser.
+   *
+   * @param userUid The UID of the user.
+   * @param messages The messages of that user
+   */
+  private fun messagesToTranscript(userUid: String, messages: List<Message>): String {
+    return messages.joinToString("\n") {
+      if (it.role == "user") "$userUid: ${it.content}" else "Interviewer: ${it.content}"
+    }
+  }
+
+  /**
+   * Evaluates the battle candidates, determines the winner of the battle and provides feedback to
+   * the winner and loser.
+   *
+   * @param user1Uid The UID of the first user.
+   * @param user2Uid The UID of the second user.
+   * @param user1messages The messages of the first user.
+   * @param user2messages The messages of the second user.
+   */
+  suspend fun evaluateBattleCandidates(
+      user1Uid: String,
+      user2Uid: String,
+      user1messages: List<Message>,
+      user2messages: List<Message>,
+      user1name: String,
+      user2name: String
+  ): EvaluationResult {
+    val user1Transcript = messagesToTranscript(user1Uid, user1messages)
+    val user2Transcript = messagesToTranscript(user2Uid, user2messages)
+
+    val prompt =
```

Comment:
#### **Enhance Prompt Maintainability in ChatViewModel**

Great work on integrating the evaluation logic! To make the ChatGPT prompts in `ChatViewModel` more maintainable and easier to update, consider extracting them into a separate `PromptTemplates` object or a dedicated constants file. This will help keep your `ChatViewModel` clean and make it simpler to manage and modify prompts in the future.

---

#### **Example Implementation**

Create a new Kotlin object to house all your prompt templates:

```kotlin
// PromptTemplates.kt
object PromptTemplates {
    fun evaluationPrompt(
        user1Uid: String,
        user2Uid: String,
        user1Transcript: String,
        user2Transcript: String
    ): String {
        return """
            You are a strict recruiter deciding between $user1Uid and $user2Uid.

            Q&A transcripts:

            $user1Uid's transcript:
            $user1Transcript

            $user2Uid's transcript:
            $user2Transcript

            Determine the winner and explain briefly why. Also, give concise feedback for the loser.
            Format:
            The winner is: <winnerUid>
            Winner message: <short message>
            Loser message: <short message>
        """.trimIndent()
    }
}
```

Update the evaluateBattleCandidates function in ChatViewModel to utilize this template:

```kotlin
val prompt = PromptTemplates.evaluationPrompt(
    user1Uid = battle.challenger,
    user2Uid = battle.opponent,
    user1Transcript = user1Transcript,
    user2Transcript = user2Transcript
)
```

Reply:
This is a great suggestion. However, it would be better to do this for all the prompts in a future PR, as it’s currently not implemented for the other prompts in the chat view model either.